moveit_core/kinematic_constraints/src/kinematic_constraint.cpp

```diff
@@ -991,7 +995,8 @@ ConstraintEvaluationResult VisibilityConstraint::decide(const robot_state::Robot
   collision_detection::CollisionRequest req;
   collision_detection::CollisionResult res;
   collision_detection::AllowedCollisionMatrix acm;
-  acm.setDefaultEntry("cone", boost::bind(&VisibilityConstraint::decideContact, this, _1));
+  acm.setDefaultEntry("cone", (boost::function<bool(collision_detection::Contact&)>)
```

Comment:
What makes this function cast necessary?

Reply:
@ahcorde  ping

Reply:
I can't explain you exactly why we have this cast. Maybe @anas can jump here and add some light. But I was trying today to change this to the current version but it was not working. My guess is because we are using C++14. I have tried to change this to `std` using `std::function` and `std::bind` because most of the packages are moving from `boost` to `std` but again I was not able to do it. 

Anyhow help is wellcome.

Reply:
I fixed this with this commit https://github.com/ros-planning/moveit2/pull/42/commits/b3855c3fb738fad92a5a8f4ae74c5068c23620aa

What do you think?

Reply:
It was causing some compilation problems, the cast prevented the problem.